src/core/Akka/Util/ByteString.cs

```diff
@@ -451,6 +458,12 @@ public byte[] ToArray()
             return copy;
         }
 
+        internal ByteBuffer ReadOnlyCompacted()
+        {
+            var c = this.Compact();
+            return c._buffers[0];
+        }
```

Comment:
The idea here is to get a Single Contiguous Arraysegment back; For smaller messages, this will most likely not copy/move anything in `.Compact()` step.

 However in either case, the ArraySegment(ByteBuffer) can then be used directly either by 'Proper' Protobuf deserializer or custom deserializer.